pandas/conftest.py

```diff
@@ -1202,6 +1203,39 @@ def utc_fixture(request):
 utc_fixture2 = utc_fixture
 
 
+@pytest.fixture(
+    params=[
+        pytest.param(None, id=str(locale.getlocale())),
+        "it_IT.utf8",
+        "zh_CN",
+    ]
+)
+def overridden_locale(request):
```

Comment:
Can use use `tm.set_locale` instead?

Reply:
I tried but I rejected it at some point as it did not seem to provide value. I had difficulties to understant the various constructs in test_localization.py, suspecting that some of these were related to localization of pandas rather than the host ?
I'll have a look again

Reply:
IIRC that helper function is supposed to be used for temporarily setting the locale in a context manager which this is trying to do to?

Reply:
Indeed. I change to this:

```python
target = request.param
if target is None:
    # Use current locale for this test.
    yield locale.setlocale(locale.LC_ALL)
else:
    if tm.can_set_locale(target, locale.LC_ALL):
        # Change locale temporarily for this test.
        with tm.set_locale(target, locale.LC_ALL):
            yield target
    else:
        # Not available on this host. Skip test.
        pytest.skip(f"Skipping as locale {repr(locale)} cannot be set.")
```

However note that `tm.can_set_locale` contains a call to `set_locale` so it is a bit less optimal. Anyway, this works and is probably more readable, I agree.

Reply:
It still seems this fixture function is very specific to the 2 added tests and pretty much calls `tm.set_locale`. Can it just be used in the testing function?

Reply:
This the reason why the fixture was in the test file, and not in conftest :)

I'll try to dispatch it

Reply:
done